### ISSUE TITLE
Add ruff linter and formatter to Spinta

### ISSUE THREAD

**Issue by sirex** [atviriduomenys/spinta#434]
Add linter and formatter:
- Linter `ruff check`
- Formatter `ruff format --check`

Together with linter and/or formatter:
- Add CI/CD check, that fails if code is linted/formatted incorrectly
- Fix lint/format problems, so that CI/CD check passes
- Include commit has in `.git-blame-ignore-revs` so that it doesn't show up in `git blame`

----

Results of `ruff format --check` as of 2025-09-12:
```
595 files would be reformatted, 291 files already formatted
```

Results of `ruff format --check` as of 2025-09-12, if using `quote-style = "single"` rule:
```
507 files would be reformatted, 379 files already formatted
```
----

Results of `ruff check` as of 2025-09-12:
```
Found 1097 errors.
[*] 389 fixable with the `--fix` option (31 hidden fixes can be enabled with the `--unsafe-fixes` option).
```

`ruff check --statistics` as of 2025-09-12:
```
651	F811	[ ] redefined-while-unused
306	F541	[*] f-string-missing-placeholders
 75	F401	[ ] unused-import
 29	F841	[*] unused-variable
 12	E722	[ ] bare-except
  7	F821	[ ] undefined-name
  6	E711	[ ] none-comparison
  4	E713	[*] not-in-test
  4	E721	[ ] type-comparison
  1	E741	[ ] ambiguous-variable-name
  1	F402	[ ] import-shadowed-by-loop-var
  1	F901	[*] raise-not-implemented
```
`F811 redefined-while-unused` rule will have to be turned off, because it doesn't work with `multipledispatch` libraries. Ignoring this rule, `ruff check` results as of 2025-09-12:
```
Found 446 errors.
[*] 383 fixable with the `--fix` option (31 hidden fixes can be enabled with the `--unsafe-fixes` option).
```
### Rules
line length: 120
quotes: double (should be doublechecked with @sirex)

### Related

- https://github.com/atviriduomenys/spinta/issues/59

**Comment by sirex** [comment]
Another option, to sue [ruff](https://github.com/astral-sh/ruff), which is fast and covers both flak8 and black.

**Comment by varzgalys** [comment]
Suggestion - introduce ruff (personal preference) but only the auto-formatting part as this can be achieved quite easily, flakness check could be implemented later or implemented as optional CI check for now, same with mypy

**Comment by Struchid** [comment]
I also agree about working with Ruff, much quicker and nicer experience than `Black + flake8 + isort` as separate dependencies.

Enabling the formatter first, and then introducing the linter part incrementally.
